spec/abstractops.html

```diff
@@ -1698,6 +1698,11 @@ <h1>
           1. Perform ! CreateDataPropertyOrThrow(_result_, _property_, _value_).
       1. If _requiredFields_ is ~partial~ and _any_ is *false*, then
         1. Throw a *TypeError* exception.
+      1. If _requiredFields_ is not ~partial~, then
+        1. Let _era_ be ? Get(_result_, *"era"*).
+        1. Let _eraYear_ be ? Get(_result_, *"eraYear"*).
+        1. If _era_ is *undefined* and _eraYear_ is not *undefined*, throw a *RangeError* exception.
+        1. If _era_ is not *undefined* and _eraYear_ is *undefined*, throw a *RangeError* exception.
```

Comment:
As per https://github.com/tc39/proposal-temporal/issues/2465#issuecomment-1397548399 we'll remove this part because it is already done in dateFromFields/yearMonthFromFields/monthDayFromFields anyway, after each call to PrepareTemporalFields.

Reply:
Is there a harm in validating it here as well?

Reply:
Yes, it "leaks" calendar-specific logic outside of Temporal.Calendar methods.

Reply:
Is there likely to be any calendar that supports one of these two properties but not both?

Reply:
No. Calendars either use eras or they won't. If they do use eras (which is, BTW, the vast majority of CLDR calendars) then `era`/`eraYear` are used as a pair. One of then without the other is not meaningful.

This means that discussions relating to validation and eras are focused on relatively narrow questions:
* In non-era-using calendars, or in 262 without 402, what should happen if callers provide (in a property bag passed to `from`, `with`, etc.) one of `era`/`eraYear` without the other?  Throw? Ignore?
* Where exactly should the validation take place that ensures that those two properties are either both present or both `undefined`?

Reply:
I think that unknown properties should be ignored, but known ones should be validated as early as possible.

Given that these two properties are really just *one* property, that contains two items, and should only come as a pair, it seems reasonable to me to validate for them here as well, since they're known property names.

Reply:
That is validated already, this just removes a duplicate validation. I get what you are saying about validating as early as possible, but I don't think that weighs against having calendar-specific logic live outside of Temporal.Calendar methods.

Reply:
This AO is called "PrepareTemporalFields" and Calendar is part of Temporal - since this is logic that would apply to any era-using calendar, it doesn't feel calendar-specific to me, it feels calendar-general, which doesn't imo need to be contained to calendar methods.